util/dtgen/helper.py

```diff
@@ -456,14 +501,17 @@ class IpHelper:
     PERIPH_IO_FIELD_NAME = Name(["periph", "io"])
     DT_STRUCT_NAME_PREFIX = Name(["dt", "desc"])
     FIRST_IRQ_FIELD_NAME = Name(["first", "irq"])
+    EXTENSION_FIELD_NAME = Name(["ext"])
 
-    def __init__(self, top_helper: TopHelper, ip: IpBlock, default_node: str,
-                 enum_type, array_mapping_type):
+    def __init__(self, top_helper: TopHelper, ip: IpBlock, ipconfig, default_node: str,
+                 enum_type, array_mapping_type, extension: Extension = None):
```

Comment:
Typing issue: `extension` is typed as non optional `Extension` and assigned `None`. It should be either `Optional[Extension]`, `Union[Extension]`, or some default `Extension` object.